homework-4/script.js

```diff
@@ -0,0 +1,247 @@
+      country.value = targetElement.getAttribute('data-country');
+
+      updateUsers.addEventListener('click', () => {
+        editElement(id);    
+      });              
+  });
+
+    tr.appendChild(checkBox);
+    tr.appendChild(tdCompanyName);
+    tr.appendChild(tdContactName);
+    tr.appendChild(tdAddress);
+    tr.appendChild(tdCity);
+    tr.appendChild(tdCountry);
+    tr.appendChild(editButton);
+    tr.appendChild(deleteButton);
+    tableBody.appendChild(tr);
+
+    tdCompanyName.innerText = element.name;
+    tdContactName.innerText = element.company;
+    tdAddress. innerText = element.address;
+    tdCity.innerText = element.city;
+    tdCountry.innerText = element.country;
+    deleteButton.innerHTML = '<i class="fa fa-close icon-close">';
+    editButton.innerHTML = '<i class="fa fa-pen icon-pen">';
+    editButton.className = 'edit';
+    deleteButton.className ='delete';
+    tdCompanyName.className ='name';
+    tr.className = 'add-tr';
+
+    // filter
+    function filterFunction(filterName, searchText) {
+      let tdName = tr.querySelectorAll('.' + filterName); 
+      for (i = 0; i < tdName.length; i++) {                
+        if (tdName[i].innerHTML.toUpperCase().indexOf(searchText.toUpperCase()) > -1) {
+          console.log(tdName);
+          tdName[i].style.display = '';
+        } else {
+          tdName[i].closest('tr').style.display = 'none';
+        }
+      }
+    }
+
+    const input = document.querySelector('.filter-button');
+    const inputFilters = document.querySelectorAll('.input');
+    input.addEventListener('click', function () {
+      let isFiltersFillded = false;
+      for (i = 0; i < inputFilters.length; i++) {
+        let val = inputFilters[i].value;
+        if (val.trim() != '') {
+          isFiltersFillded = true;  
+          let filterName = inputFilters[i].getAttribute('data-filter');
+          filterFunction(filterName, val.trim());
+        }
+      }
+      
+      if (!isFiltersFillded) {
+        showAll();
+      }
+    });
+
+    // Show all rows.
+    function showAll() {
+      let trs = document.querySelectorAll('tr.add-tr');
+      for (i = 0; i < trs.length; i++) {
+        trs[i].style.display = '';
+      }
+    }
+  }
+
+  // Edit element
+  function editElement(id) {
+    fetch(`${baseUrl}/posts/${id}`, {
+      method: 'PUT',
+      headers: {
+        "Content-type": "application/json",
+      },
+      body: JSON.stringify({
+        company: companyName.value,
+        name: contactName.value,
+        address: address.value,
+        city: city.value,
+        country: country.value,
+      }),
+    })
+    .then(data => data.json())
+    .then(res => res);
+  }
+
+  // Delete element.
+  function deleteElement(id) {
+    fetch(`${baseUrl}/posts/${id}`, {
+      method: 'DELETE',
+    })
+    .then(data => data.json())
+    .then(res => res);
+  }
+
+  modalClose.addEventListener('click', (e) => {
```

Comment:
you have two different modals. Looking at this code it is impossible to understand what modal will be closed.

Anyway you don't need two modals - use one for adding and editing cutomers.